File: Sources/CmuxPetKit/Voice/Wording.swift
// Texto que produce el PROGRAMA, no la mascota.
//
// Nada de aqui tiene personalidad, a proposito: si el programa pusiera
// onomatopeyas o muletillas, un gato acabaria diciendo "*whirr*". La personalidad
// vive en el pack; esto son piezas neutras que las plantillas del pack rellenan,
// mas el ultimo recurso cuando una mascota no trae frase para algo.
//
// Ver docs/adr/0005.

import Foundation

public enum Wording {
    /// Traduce la herramienta a lo que el agente esta haciendo de verdad.
    /// `tool_input` viene redactado en los eventos, asi que el nombre de la
    /// herramienta es todo lo que hay: sirve para el verbo, no para el detalle.
    public static func activity(_ tool: String?) -> String {
        switch tool ?? "" {
        case "Bash", "BashOutput", "KillShell":            return "corriendo comandos"
        case "Edit", "Write", "MultiEdit", "NotebookEdit":  return "editando archivos"
        case "Read":                                        return "leyendo código"
        case "Grep", "Glob":                                return "buscando en el código"
        case "WebFetch", "WebSearch":                       return "consultando la web"
        case "Task", "Agent":                               return "delegando a subagentes"
        case "Skill":                                       return "usando una skill"
        case "AskUserQuestion":                             return "esperando tu respuesta"
        case "TodoWrite", "TaskCreate", "TaskUpdate":       return "organizando el plan"
        case "Artifact":                                    return "publicando un artifact"
        case "":                                            return "arrancando"
        default:                                            return "usando \(tool ?? "")"
        }
    }

    /// " en Fineract" o cadena vacia si no sabemos donde. Trae la preposicion
    /// incluida para que las plantillas lo peguen directo tras una palabra.
    public static func at(_ workspace: String) -> String {
        workspace.isEmpty ? "" : " en \(workspace)"
    }

    /// "y 2 más": lo que se agrega cuando la frase de la mascota habla de un
    /// agente pero hay varios. Neutro, para no pisar su voz.
    public static func andOthers(_ count: Int) -> String {
        count <= 0 ? "" : (count == 1 ? " Y otro más." : " Y otros \(count) más.")
    }

    /// Ultimo recurso: la mascota no trae frase para esa clase y todavia no se le
    /// han generado. Informa sin actuar ningun personaje.
    public static func plain(_ text: String) -> String { text }
}

File: windows/cmux_pet_win/wording.py
# Texto que produce el PROGRAMA, no la mascota. Puerto de Voice/Wording.swift.
#
# Nada de aqui tiene personalidad, a proposito: la personalidad vive en el pack.
# Esto son piezas neutras que las plantillas del pack rellenan.

# Traduce la herramienta a lo que el agente esta haciendo de verdad. En Windows
# los hooks de Claude Code SI traen el nombre real de la herramienta (a
# diferencia de cmux, que lo redacta), asi que el verbo es fiable.
_ACTIVITY = {
    "Bash": "corriendo comandos",
    "BashOutput": "corriendo comandos",
    "KillShell": "corriendo comandos",
    "Edit": "editando archivos",
    "Write": "editando archivos",
    "MultiEdit": "editando archivos",
    "NotebookEdit": "editando archivos",
    "Read": "leyendo codigo",
    "Grep": "buscando en el codigo",
    "Glob": "buscando en el codigo",
    "WebFetch": "consultando la web",
    "WebSearch": "consultando la web",
    "Task": "delegando a subagentes",
    "Agent": "delegando a subagentes",
    "Skill": "usando una skill",
    "AskUserQuestion": "esperando tu respuesta",
    "TodoWrite": "organizando el plan",
    "TaskCreate": "organizando el plan",
    "TaskUpdate": "organizando el plan",
    "Artifact": "publicando un artifact",
    "": "arrancando",
}


def activity(tool: str) -> str:
    tool = tool or ""
    if tool in _ACTIVITY:
        return _ACTIVITY[tool]
    return f"usando {tool}"


def at(workspace: str) -> str:
    """ ' en Fineract' o cadena vacia. Trae la preposicion incluida. """
    return f" en {workspace}" if workspace else ""


def and_others(count: int) -> str:
    if count <= 0:
        return ""
    return " Y otro mas." if count == 1 else f" Y otros {count} mas."


def update_available(version: str) -> str:
    """ Ultimo recurso cuando el pack no trae frase para la clase. """
    return f"Hay una version nueva de cmux-pet: {version}. Corre: python install.py --update"


def fmt_duration(seconds: float) -> str:
    """ '1 min 34 s', '12 s', '2 h 3 min'. Duracion ya formateada para {time}. """
    s = int(round(seconds))
    if s < 60:
        return f"{s} s"
    if s < 3600:
        m, r = divmod(s, 60)
        return f"{m} min {r} s" if r else f"{m} min"
    h, r = divmod(s, 3600)
    m = r // 60
    return f"{h} h {m} min" if m else f"{h} h"
